tools/sync_tron_state_from_k8s.py

```diff
@@ -0,0 +1,202 @@
+    "unknown",
+    "lost",
+]
+
+log = logging.getLogger("sync_tron_from_k8s")
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--kubeconfig-path", dest="kubeconfig_path", help="KUBECONFIG path")
+    parser.add_argument(
+        "--do-work",
+        dest="do_work",
+        action="store_true",
+        default=False,
+        help="Actually modify tron actions that need updating; without this flag we will only print those that would be updated",
+    )
+    parser.add_argument("--tron-url", default=None, help="Tron url (default will read from paasta tron config)")
+    parser.add_argument(
+        "--tronctl-wrapper",
+        default="tronctl",
+        dest="tronctl_wrapper",
+        help="Tronctl wrapper to use (will not use wrapper by default)",
+    )
+    parser.add_argument("-n", "--num-runs", dest="num_runs", default=100, help="Maximum number of job runs to retrieve")
+    parser.add_argument("-v", "--verbose", dest="verbose", action="store_true", default=False, help="Verbose logging")
+    args = parser.parse_args()
+
+    # tron's base level is critical, not info, adjust accoringly
+    if args.verbose:
+        level = logging.DEBUG
+        tron_level = logging.WARN
+    else:
+        level = logging.INFO
+        tron_level = logging.CRITICAL
+
+    logging.basicConfig(level=level, stream=sys.stdout)
+
+    tron_client_logger = logging.getLogger("tron.commands.client")
+    tron_client_logger.setLevel(tron_level)
+
+    # We also don't want kube_client debug logs
+    kube_logger = logging.getLogger("kubernetes.client.rest")
+    kube_logger.setLevel(logging.INFO)
+
+    return args
+
+
+def fetch_pods(kubeconfig_path: str) -> Dict[str, V1Pod]:
+    kube_client = KubeClient(kubeconfig_path=kubeconfig_path, user_agent="sync_tron_state_from_k8s")
+
+    # Bit of a hack, no helper to fetch pods so reach into core api
+    completed_pod_list = kube_client.core.list_namespaced_pod(
+        namespace="tron",
+    )
+
+    return {pod.metadata.name: pod for pod in completed_pod_list.items}
+
+
+def get_tron_state_from_api(tron_server: str, num_runs: int = 100) -> List[Dict[str, Dict[Any, Any]]]:
+    if not tron_server:
+        client_config = get_client_config()
+        tron_server = client_config.get("server", "http://localhost:8089")
+    client = Client(tron_server)
+    # /jobs returns only the latest 5 runs, we'll need to request all runs instead ourselves
+    jobs = client.jobs(
+        include_job_runs=False,
+        include_action_runs=False,
+        include_action_graph=False,
+        include_node_pool=False,
+    )
+
+    for job in jobs:
+        # What am I doing wrong here, why do I have to append /api
+        url = f'/api{job["url"]}'
+        log.debug(f'Fetching job {job["name"]} at {url}')
+        job_runs = client.job(
+            url,
+            include_action_runs=True,
+            count=num_runs,  # TODO: fetch job run_limit and use that for count ?
```

Comment:
not a bad idea :)

Reply:
Turns out we don't actually include run_limit in job/job run API responses :/  so this would need to pull in from other sources (either config api or reading the configs some other way), so I think I'm going to leave it as is for now (we can always run w/ a higher num_run if we need to).

Reply:
yea, we can probably read from /nail/etc/services - but it's probably fine to leave as-is, i can't think of anything getting anywhere near their `run_limit` in terms of concurrently running jobs